pkg/healthcheck/healthcheck.go

```diff
@@ -499,3 +516,46 @@ func validateDataPlanePods(pods []v1.Pod, targetNamespace string) error {
 
 	return nil
 }
+
+func validateDataPlanePodReporting(k8sPods []v1.Pod, promPods []*pb.Pod, targetNamespace string) error {
```

Comment:
If you feel inclined, I think it would be pretty straightforward to add unit tests for this method in `pkg/healthcheck/healthcheck_test.go`. If you look in that file you'll see existing unit tests for the rest of the `validate*` funcs in this file.